x/lscosmos/keeper/allowlisted_validators.go

```diff
@@ -20,3 +20,69 @@ func (k Keeper) GetAllowListedValidators(ctx sdk.Context) types.AllowListedValid
 
 	return allowListedValidators
 }
+
+// GetAllValidatorsState returns the combined allowed listed validators set and combined
+// delegation state. It is done to keep the old validators in the loop while calculating weighted amounts
+// for delegation and undelegation
+func (k Keeper) GetAllValidatorsState(ctx sdk.Context) (types.AllowListedValidators, types.DelegationState) {
```

Comment:
where is the test for this function?

Reply:
Tests will be added later. It was needed urgently.

Reply:
hard to figure out what the code is doing without tests

Reply:
Added test file.